kivy/__init__.py

```diff
@@ -380,6 +380,16 @@ def _patch_mod_deps_win(dep_mod, mod_name):
     level = LOG_LEVELS.get(Config.get('kivy', 'log_level'))
     Logger.setLevel(level=level)
 
+    # Initialize our logger, if we do not log anything during this
+    # init sequence then the logger will not be `_configured`. This
+    # can occure when log level is set to a stricter level like `error`
+    # and no errors occur, then when we call the purge_logs function our
+    # handler could be un-configured and not have a log_dir attribute --
+    # resulting in the function bailing out.
+    from kivy.logger import file_log_handler
+    file_log_handler._configure()
```

Comment:
`file_log_handler` may be `None`, so we should check if is not, before calling `_configure`, as we do just before the call of `purge_logs`.

Reply:
I am not sure if it can be done exactly on initialization since it looks like importing config creates a circular import error:

```python
Traceback (most recent call last):
  File "/Users/craigkaiser/kaiserengineering/digitaldash/main.py", line 20, in <module>
    from kivy.logger import Logger, LOG_LEVELS
  File "/Users/craigkaiser/.local/share/virtualenvs/digitaldash-iuTs6a7o/lib/python3.11/site-packages/kivy/__init__.py", line 40, in <module>
    from kivy.logger import Logger, LOG_LEVELS
  File "/Users/craigkaiser/.local/share/virtualenvs/digitaldash-iuTs6a7o/lib/python3.11/site-packages/kivy/logger.py", line 634, in <module>
    add_kivy_handlers(logging.root)
  File "/Users/craigkaiser/.local/share/virtualenvs/digitaldash-iuTs6a7o/lib/python3.11/site-packages/kivy/logger.py", line 598, in add_kivy_handlers
    from kivy.config import Config
  File "/Users/craigkaiser/.local/share/virtualenvs/digitaldash-iuTs6a7o/lib/python3.11/site-packages/kivy/config.py", line 387, in <module>
    from kivy import kivy_config_fn
ImportError: cannot import name 'kivy_config_fn' from partially initialized module 'kivy' (most likely due to a circular import) (/Users/craigkaiser/.local/share/virtualenvs/digitaldash-iuTs6a7o/lib/python3.11/site-packages/kivy/__init__.py)
```